src/main/java/rx/observers/TestSubscriber.java

```diff
@@ -670,4 +673,33 @@ final void assertionError(String message) {
         }
         throw ae;
     }
+    
+    /**
+     * Assert that the TestSubscriber contains the given first and optional rest values exactly
+     * and if so, clears the internal list of values.
+     * <p>
+     * <code><pre>
+     * TestSubscriber ts = new TestSubscriber();
+     * 
+     * ts.onNext(1);
+     * 
+     * ts.assertAndConsume(1);
+     * 
+     * ts.onNext(2);
+     * ts.onNext(3);
+     * 
+     * ts.assertAndConsume(2, 3); // no mention of 1
+     * </pre></code>
+     * @param expectedFirstValue the expected first value
+     * @param expectedRestValues the optional rest values
+     */
+    public final void assertAndConsume(T expectedFirstValue, T... expectedRestValues) {
```

Comment:
What about assertValuesAndClear()?

On Wed, 10 Aug 2016, 20:05 David Karnok, <notifications@github.com> wrote:

> In src/main/java/rx/observers/TestSubscriber.java
> <https://github.com/ReactiveX/RxJava/pull/4322#discussion_r74286042>:
>
> > +     * <code><pre>
> > +     * TestSubscriber ts = new TestSubscriber();
> > +     *
> > +     * ts.onNext(1);
> > +     *
> > +     * ts.assertAndConsume(1);
> > +     *
> > +     * ts.onNext(2);
> > +     * ts.onNext(3);
> > +     *
> > +     * ts.assertAndConsume(2, 3); // no mention of 1
> > +     * </pre></code>
> > +     * @param expectedFirstValue the expected first value
> > +     * @param expectedRestValues the optional rest values
> > +     */
> > +    public final void assertAndConsume(T expectedFirstValue, T... expectedRestValues) {
>
> That implies the whole internal state is wiped out and you want to reuse
> the same instance on another Observable.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/4322/files/5429edf090632abfd0da6a5f00fa13da122e5650#r74286042>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA7B3GsAbSwdMB9RM6jECO8uica-3cClks5qegS9gaJpZM4Jf7p4>
> .
>